### PR TITLE
[FIX] account: do not use optional library

### PR DESCRIPTION
**Pull request by mart-e**
stdnum is not a mandatory dependency
It is often present as required to install the base_vat module but
having a system with account without the dependency should be
possible.

As the calc_check_digits is pretty small, it is easy to extract

Replaces and closes odoo/odoo#40067

### CODE REVIEW

File: addons/account/models/account_invoice.py [hunk 1]
```diff
@@ -40,6 +39,18 @@
...
 MAGIC_COLUMNS = ('id', 'create_uid', 'create_date', 'write_uid', 'write_date')
 
+def calc_check_digits(number):
+    """Calculate the extra digits that should be appended to the number to make it a valid number.
+    Source: python-stdnum iso7064.mod_97_10.calc_check_digits
+    """
+    try:
+        number_base10 = ''.join(str(int(x, 36)) for x in number)
+    except Exception:
+        raise UserError(_("Invalid format for %s") % number)
```
Comment (xmo-odoo) [inline addons/account/models/account_invoice.py]: The old code didn't convert from internal stdnum exceptions to Odoo exceptions so maybe we can just let the ValueError bubble up if a non-alphanumeric IBAN is provided?

**Comment by mart-e** [comment]
@smetl thanks but too fast, still had an issue with error message ;-)

File: addons/account/models/account_invoice.py [hunk 2]
```diff
@@ -40,6 +39,18 @@
...
+
+    checksum = int(number_base10) % 97
+    return '%02d' % ((98 - 100 * checksum) % 97)
```
Comment (smetl) [inline addons/account/models/account_invoice.py]: Is `return '%02d' % ((98 - 100 * checksum(number)) % 97)` not enought?

File: addons/account/models/account_invoice.py (hunk 1 above)
Comment (mart-e) [reply addons/account/models/account_invoice.py]: right, removing the try/except

File: addons/account/models/account_invoice.py (hunk 2 above)
Comment (mart-e) [reply addons/account/models/account_invoice.py]: but `checksum` is also a oneliner of stdnum which itself called another oneliner `_to_base10` ;-)
https://github.com/arthurdejong/python-stdnum/blob/831c66990ab8fb3040e5b7fc74668836a1d4ef64/stdnum/iso7064/mod_97_10.py#L40-L51

**Comment by mart-e** [comment]
robodoo r+

**Pull request closed without merging**